src/mvs_eland/C0_data_processing.py

```diff
@@ -1696,35 +1699,49 @@ def add_maximum_cap(dict_values, group, asset, subasset=None):
     asset: str
         asset name
     subasset: str
-        subasset name
+        subasset name.
+        Default: None.
 
     Returns
     -------
 
     """
     if subasset is None:
-        dict = dict_values[group][asset]
+        asset_dict = dict_values[group][asset]
     else:
-        dict = dict_values[group][asset][subasset]
-    if MAXIMUM_CAP in dict:
-        # check if maximumCap is greater that installedCap
-        if dict[MAXIMUM_CAP][VALUE] is not None:
-            if dict[MAXIMUM_CAP][VALUE] < dict[INSTALLED_CAP][VALUE]:
+        asset_dict = dict_values[group][asset][subasset]
+    if MAXIMUM_CAP in asset_dict:
+        # Check if a maximumCap is defined
+        if asset_dict[MAXIMUM_CAP][VALUE] is not None:
+            # adapt maximumCap of non-dispatchable sources
+            if (
+                group == ENERGY_PRODUCTION
+                and asset_dict[FILENAME] != None
+                and max(asset_dict[TIMESERIES_NORMALIZED]) != 1
+            ):
+                asset_dict[MAXIMUM_CAP][VALUE] = asset_dict[MAXIMUM_CAP][VALUE] / max(
+                    asset_dict[TIMESERIES_NORMALIZED]
```

Comment:
I do not understand how the `max(normalized timeseries)` can be `!=1`. You defined `TIMESERIES_NORMALIZED `by deviding the `TIMESERIES `by its `TIMESERIES_PEAK`. `max(TIMESERIES_NORMALIZED)` therefore must be `==1`. I think you should check for `TIMESERIES_PEAK `here and divide the `MAXIMUM_CAP `by that.

Reply:
Right!

Do you mean `and asset_dict[TIMESERIES_PEAK] != None` ? But doesn't every time series have a TIMESERIES_PEAK?

Reply:
There might be DSO sources that do not have the `TIMESERIES_PEAK`. To be sure nothing weird happens, we can just check that it exists:
```
...and TIMESERIES_PEAK in asset_dict
     asset_dict[MAXIMUM_CAP][VALUE] = asset_dict[MAXIMUM_CAP][VALUE] / max(
                    asset_dict[TIMESERIES_PEAK]
else:
    logging.warning(f`The asset {asset_dict[LABEL]} does not have a timeseries peak defined. The maximumCap is set to None.`)
```

Reply:
Ah okay! 
But a DSO source wouldn't have asset_dict[FILENAME] - with that we already check for only non-dispatchable sources.
We could still check for `TIMESERIES_PEAK in asset_dict` to be sure!

Reply:
Ah, right!